Flaky test `aiohttp.client_exceptions.ClientOSError: [Errno 54] Connection reset by peer`

```
============================= test session starts ==============================
platform darwin -- Python 3.9.7, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /Users/runner/work/dd-apm-test-agent/dd-apm-test-agent, configfile: setup.cfg, testpaths: ddapm_test_agent, tests
plugins: ddtrace-0.51.1
collected 62 items

ddapm_test_agent/agent.py .                                              [  1%]
ddapm_test_agent/snapshot.py ..                                          [  4%]
tests/test_agent.py ..                                                   [  8%]
tests/test_checks.py .......                                             [ 19%]
tests/test_session.py .......                                            [ 30%]
tests/test_snapshot.py ...............                                   [ 54%]
tests/test_snapshot_integration.py ..E.........                          [ 74%]
tests/test_trace.py ................                                     [100%]

==================================== ERRORS ====================================
_ ERROR at setup of test_single_trace[pyloop-root-custom_service2-/url/endpoint-0-web-meta2-metrics2-400] _

loop = <_UnixSelectorEventLoop running=False closed=False debug=False>
testagent_port = 8126

    @pytest.fixture
    async def testagent(loop, testagent_port):
        env = os.environ.copy()
        env.update(
            {
                "PORT": str(testagent_port),
                "SNAPSHOT_CI": "1",
                "SNAPSHOT_DIR": os.path.join(
                    os.path.dirname(__file__), "integration_snapshots"
                ),
            }
        )
        p = subprocess.Popen(["ddapm-test-agent"], env=env)
    
        # Wait for server to start
        try:
            async with aiohttp.ClientSession() as session:
                for _ in range(20):
                    try:
>                       r = await session.get(f"http://localhost:{testagent_port}")

tests/test_snapshot_integration.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.riot/venv_py397/lib/python3.9/site-packages/aiohttp/client.py:544: in _request
    await resp.start(conn)
.riot/venv_py397/lib/python3.9/site-packages/aiohttp/client_reqrep.py:890: in start
    message, payload = await self._protocol.read()  # type: ignore
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <aiohttp.client_proto.ResponseHandler object at 0x1056bc640>

    async def read(self) -> _T:
        if not self._buffer and not self._eof:
            assert not self._waiter
            self._waiter = self._loop.create_future()
            try:
>               await self._waiter
E               aiohttp.client_exceptions.ClientOSError: [Errno 54] Connection reset by peer

.riot/venv_py397/lib/python3.9/site-packages/aiohttp/streams.py:604: ClientOSError
=============================== warnings summary ===============================
tests/test_agent.py: 2 warnings
tests/test_checks.py: 7 warnings
tests/test_session.py: 7 warnings
tests/test_snapshot.py: 14 warnings
  /Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/asyncio/base_events.py:1518: DeprecationWarning: The loop argument is deprecated since Python 3.8, and scheduled for removal in Python 3.10.
    await tasks.sleep(0, loop=self)

```

--- a/tests/test_snapshot_integration.py
+++ b/tests/test_snapshot_integration.py
@@ -4,6 +4,7 @@ import subprocess
 
 import aiohttp
 from aiohttp.client_exceptions import ClientConnectorError
+from aiohttp.client_exceptions import ClientOSError
 from ddtrace import Tracer
 import pytest
 
@@ -33,7 +34,7 @@ async def testagent(loop, testagent_port):
             for _ in range(20):
                 try:
                     r = await session.get(f"http://localhost:{testagent_port}")
-                except ClientConnectorError:
+                except (ClientConnectorError, ClientOSError):
                     pass
                 else:
                     if r.status == 404:

